backend/tests/integration/accounts/test_views.py

```diff
@@ -225,3 +225,18 @@ def test_orcid_push_setting_without_user(inspire_app):
 
     expected_status = 401
     assert expected_status == response.status_code
+
+
+def test_set_email_happy_flow(inspire_app):
+    orcid = "0000-0001-8829-5461"
+    new_email = "test@cern.ch"
+    user = create_user(role="user", orcid=orcid)
+    with inspire_app.test_client() as client:
+        login_user_via_session(client, email=user.email)
+        response = client.post(
+            "/accounts/settings/update-email",
+            data=orjson.dumps({"new_email": new_email}),
+            content_type="application/json",
+        )
+    assert response.status_code == 200
+    assert user.email == new_email
```

Comment:
can we add it as async and query db directly?

Reply:
Do we really need it async? It doesn't seem like a very complex operation and it won't be used super often probably.
`query db directly?` -> isn't it what I did?

Reply:
Since we don't commit any changes on integration, it's better to test it on async to make sure the change is in the db too.